internal/objectstore/base.go

```diff
@@ -53,13 +53,15 @@ type opType int
 
 const (
 	opGet opType = iota
+	opGetByHash
 	opPut
 	opRemove
 )
 
 type request struct {
 	op            opType
 	path          string
+	sha256Prefix  string
```

Comment:
I'm starting to think this should have embedded types here.

I'll swing around to this later.